proxy/agent.go

```diff
@@ -78,6 +78,12 @@ var (
 	}
 )
 
+const (
+	// MaxAborts is the maximum number of cascading abort messages to buffer.
+	// This should be the upper bound on the number of proxies available at any point in time.
+	MaxAborts = 10
```

Comment:
Maybe limit MaxAborts to 4 (i.e. max envoy instances +1). Otherwise it looks like an arbitrary constant that we're guessing should fix the problem.